backend/session/logind.c

```diff
@@ -274,7 +274,7 @@ static int pause_device(sd_bus_message *msg, void *userdata,
 		goto error;
 	}
 
-	if (major == DRM_MAJOR) {
+	if (major == DRM_MAJOR && strcmp(type, "gone") != 0) {
```

Comment:
Maybe we should only set active = false if `strcmp(type, "pause") == 0`?

Reply:
I can test that tonight if you want

Reply:
Yeah, please do.

We might want to destroy the sub-backend if the device is gone.

Reply:
Your way seems to work as well, the following log has some debug code from @RedSoxFan as well.

https://mthode.org/dist/logs/tb-disconnect-working.log.xz

Reply:
> Maybe we should only set active = false if `strcmp(type, "pause") == 0`?

Looking at the documentation at https://www.freedesktop.org/wiki/Software/systemd/logind/ (sorry can't link to an anchor since the id's are not unique...), it looks like there are three values: `force`, `pause`, and `gone`. `force` is an asynchronous notification that the device was paused, `pause` is a synchronous notification the device is being paused, and `gone` is a notification that the device has been removed.

Based on that, I think that the condition is correct as-is. Thoughts?

> We might want to destroy the sub-backend if the device is gone.

How would I go about doing that? I haven't really touched wlroots much

Reply:
>Based on that, I think that the condition is correct as-is. Thoughts?

Indeed, you're right!

>How would I go about doing that? I haven't really touched wlroots much

`wlr_backend_destroy` will probably do the trick.

Reply:
So, this pr is invalid because it's correct as is?  What should be done to address the lock up issue then?

Reply:
No, this PR LGTM.